syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.server";
option java_outer_classname = "ServerProto";
option objc_class_prefix = "HLW";

package server;

// The Session service definition.
service SessionManager {
  rpc create (newSession) returns (Session) {}
  rpc join (JoinRequest) returns (Session) {}
  rpc leave (LeaveRequest) returns (LeaveReply) {}
  rpc setMax (SetMaxPlayers) returns (Session) {}
  rpc list (ListRequest) returns (ListReply) {}
}

message newSession {
  string name = 1;
}

message ListRequest {
  uint32 limit = 1;
}

message ListReply {
  repeated Session sessions = 1;
}

message LeaveRequest {
  string sessionId = 1;
}

message LeaveReply {
  bool success = 1;
}

message SetMaxPlayers {
  uint32 number = 1;
}

message JoinRequest {
  string sessionId = 1;
}

message Session {
  string sessionId = 1;
  string name = 2;
  //required string 
}